### PR TITLE
feat: item page translations behind login

### PR DESCRIPTION
**Pull request by LeoniePeters**

### CODE REVIEW

File: packages/portal/src/components/item/ItemLanguageSelector.vue [hunk 1]
```diff
@@ -19,6 +19,21 @@
...
               />
+              <i18n
+                v-else-if="!$auth.loggedIn"
+                path="multilingual.loginToTranslate"
+                tag="span"
+                data-qa="translate item login suggestion"
+              >
+                <template #login>
+                  <b-link
+                    data-qa="log in button"
```
Comment (rwd) [inline packages/portal/src/components/item/ItemLanguageSelector.vue:30]: This "link" can't be opened in a new tab. Make it a real link like the main log in (in top nav).

File: packages/portal/src/lang/en.js [hunk 2]
```diff
@@ -1099,6 +1101,10 @@ export default {
     },
     "results": {
       "limitWarning": "Additional results are not shown as only the first 1000 most relevant results are shown. If you haven't found what you're looking for, please consider refining your search.",
+      "login": "Do not translate",
+      "loginToSeeMore": "{login} Do not translate",
+      "loginToSeeMultilingualResults": "Log in to search across the entire collection, and get results not limited to the language of your search term.",
+      "showingMultilungualResults": "Showing results across the entire collection and all languages.",
```
Comment (rwd) [inline packages/portal/src/lang/en.js:1107]: This should be `showingMultilingualResults`

**Review by rwd — CHANGES_REQUESTED** [review]

File: packages/portal/src/lang/en.js [hunk 3]
```diff
@@ -1099,6 +1101,10 @@ export default {
     },
     "results": {
       "limitWarning": "Additional results are not shown as only the first 1000 most relevant results are shown. If you haven't found what you're looking for, please consider refining your search.",
+      "login": "Do not translate",
+      "loginToSeeMore": "{login} Do not translate",
```
[the remark this answers is not in the document]
Comment (LeoniePeters) [reply packages/portal/src/lang/en.js:1105]: No. It was meant to clarify these strings only need translations for Spanish. We still need the English strings as they are the default and translation source. That's updated.

File: packages/portal/src/pages/search/index.vue [hunk 4]
```diff
@@ -94,6 +95,9 @@
       searchOverrides() {
         const sort = 'score desc,contentTier desc,random_europeana asc,timestamp_update desc,europeana_id asc';
         return !this.searchQuery && !this.$route.query.sort ? { sort } : {};
+      },
+      doNotTranslate() {
+        return !this.$auth.loggedIn && this.$i18n.locale === 'es';
```
[the remark this answers is not in the document]
Comment (LeoniePeters) [reply packages/portal/src/pages/search/index.vue:100]: When not logged in and the locale is set to Spanish (or as defined in `$config.app.search.translateLocales`) the prop is set to true. The `doNotTranslate` prop is then used in [SearchInterface](https://github.com/europeana/portal.js/blob/f29e740bc17b9af6e755405c94d1527920dcf105/packages/portal/src/components/search/SearchInterface.vue#L373-L386) to determine whether to set the `translateLang` api option used to construct the search API query.
The `doNotTranslate` prop was already introduced to disable translations on collection pages. So I reused it for this case too.
I've added a comment in the code for future reference.

File: packages/portal/src/components/item/ItemLanguageSelector.vue (hunk 1 above)
Comment (LeoniePeters) [reply packages/portal/src/components/item/ItemLanguageSelector.vue:30]: Done.

File: packages/portal/src/lang/en.js (hunk 2 above)
Comment (LeoniePeters) [reply packages/portal/src/lang/en.js]: Updated.

File: packages/portal/src/components/item/ItemLanguageSelector.vue [hunk 5]
```diff
@@ -19,6 +19,21 @@
...
+                    @click="keycloakLogin"
+                  >
+                    {{ $t('multilingual.login') }}
+                  </b-link>
+                </template>
+              </i18n>
```
[the remark this answers is not in the document]
Comment (LeoniePeters) [reply packages/portal/src/components/item/ItemLanguageSelector.vue:38]: Fixed.

File: packages/portal/src/pages/item/_.vue [hunk 6]
```diff
@@ -328,6 +328,13 @@
       }
     },
 
+    created() {
+      // When entering a translated item page, but not logged in, redirect to non-translated item page
```
[the remark this answers is not in the document]
Comment (LeoniePeters) [reply packages/portal/src/pages/item/_.vue]: Moved to `fetch`.

File: packages/portal/src/pages/search/index.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (LeoniePeters) [reply packages/portal/src/pages/search/index.vue]: And simplified it to only check the logged in state on search pages. On collection pages it's updated to check the logged in state plus the config's `app.search.collections.doNotTranslate` state.
The `app.search.translateLocales` state is checked late on in the [translateLang](https://github.com/europeana/portal.js/blob/4dd9823f4f3a3428c9108a8ad9bd970ba32f31c4/packages/portal/src/components/search/SearchInterface.vue#L381) prop

File: packages/portal/src/pages/item/_.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (rwd) [reply packages/portal/src/pages/item/_.vue]: This is still not preventing the API request being made. Observe both console logs being made if one is not logged-in and requests a translated page, with this patch applied:

```diff
diff --git a/packages/portal/src/pages/item/_.vue b/packages/portal/src/pages/item/_.vue
index 9fa445bb6..fe26e79a6 100644
--- a/packages/portal/src/pages/item/_.vue
+++ b/packages/portal/src/pages/item/_.vue
@@ -212,8 +212,10 @@
     async fetch() {
       // When entering a translated item page, but not logged in, redirect to non-translated item page
       if (this.$route.query.lang && !this.$auth.loggedIn) {
+        console.log('fetch redirect w/o lang')
         this.redirectToAltRoute({ query: { lang: undefined } });
       }
+      console.log('fetch metadata')
 
       await this.fetchMetadata();
     },
```

[the remark this answers is not in the document]
Comment (LeoniePeters) [reply packages/portal/src/pages/item/_.vue]: Indeed. That should be fixed now.

File: packages/portal/src/components/search/SearchResultsContext.vue [hunk 7]
```diff
@@ -41,7 +45,32 @@
...
+ -->
+    <i18n
+      v-if="suggestLoginForMoreResults"
+      path="search.results.loginToSeeMore"
+      tag="span"
+      class="context-label mr-1"
+      data-qa="results more link"
+    >
+      <template #login>
+        <b-link
```
[the remark this answers is not in the document]
Comment (LeoniePeters) [reply packages/portal/src/components/search/SearchResultsContext.vue:58]: Fixed to work the same as the top navigation and item page login links.

**Pull request merged**